lib/translate/inquiry.cc

```diff
@@ -0,0 +1,313 @@
+  result.peripheral_qualifier =
+      scsi_defs::PeripheralQualifier::kPeripheralDeviceConnected;
+  result.peripheral_device_type =
+      scsi_defs::PeripheralDeviceType::kDirectAccessBlock;
+  result.rmb = 0;
+  result.version = scsi_defs::Version::kSpc4;
+  result.normaca = 0;
+  result.hisup = 0;
+  result.response_data_format = scsi_defs::ResponseDataFormat::kCompliant;
+  result.additional_length = 0x1f;
+  result.sccs = 0;
+  result.acc = 0;
+  result.tpgs = scsi_defs::TPGS::kNotSupported;
+  result.third_party_copy = 0;
+  result.protect = (identify_namespace_data.dps.pit == 0 &&
+                    identify_namespace_data.dps.md_start == 0)
+                       ? 0
+                       : 1;
+  result.encserv = 0;
+  result.multip = 0;
+  result.addr_16 = 0;
+  result.wbus_16 = 0;
+  result.sync = 0;
+  result.cmdque = 1;
+
+  // Shall be set to “NVMe” followed by 4 spaces: “NVMe “
+  strncpy(result.vendor_identification, NVME_VENDOR_IDENTIFICATION, 8);
+
+  // Shall be set to the first 16 bytes of the Model Number (MN) field within
+  // the Identify Controller Data Structure
+  memcpy(result.product_identification, identify_controller_data.mn, 16);
+
+  /*
+  Shall be set to the last 4 ASCII graphic characters in the range of 21h-7Eh
+  (i.e. last 4 non-space characters) of the Firmware Revision (FR) field within
+  the Identify Controller Data Structure.
+  */
+  int idx = 3;
+  for (int i = 7; i >= 0; i--) {
+    if (identify_controller_data.fr[i] != ' ' &&
+        identify_controller_data.fr[i] >= 0x21 &&
+        identify_controller_data.fr[i] <= 0x7e) {
+      result.product_revision_level[idx] = identify_controller_data.fr[i];
+      if (idx-- == 0) break;
+    }
+  }
+
+  if (idx >= 0) {
+    char debug_buffer[100];
+    sprintf(debug_buffer, "less than four characters set\n");
+    if (debug_callback != NULL) debug_callback(debug_buffer);
+  }
+  return result;
+}
+
+scsi_defs::InquiryData BuildStandardInquiry() {
+  // Identify controller results
+  nvme_defs::IdentifyControllerData identify_controller_data =
+      NvmeIdentifyController();
+  // Identify namespace results
+  nvme_defs::IdentifyNamespace identify_namespace_data =
+      NvmeIdentifyNamespace();
+  return TranslateStandardInquiryResponse(identify_controller_data,
+                                          identify_namespace_data);
+}
+
+scsi_defs::SupportedVitalProductData BuildSupportedVpdPages() {
+  scsi_defs::SupportedVitalProductData result =
+      scsi_defs::SupportedVitalProductData();
+  result.peripheral_qualifier =
+      scsi_defs::PeripheralQualifier::kPeripheralDeviceConnected;
+  result.peripheral_device_type =
+      scsi_defs::PeripheralDeviceType::kDirectAccessBlock;
+  result.page_code = 0;
+
+  // Shall be set to 5 indicating number of items supported Vpd pages list
+  // requires. NOTE: document says to set this to 5 but there are 7 entries....
+  result.page_length = 5;
+
+  result.supported_page_list[0] = 0x00;
```

Comment:
We can initialize this in single line with `.supported_page_list = {0x00, 0x80... ` and we should use enum here as well.